app/graphql/resolvers/submission_resolver.rb

```diff
@@ -1,19 +1,20 @@
 # frozen_string_literal: true
 
 class SubmissionResolver < BaseResolver
-  def valid?
-    unless admin?
-      bad_argument_error =
-        GraphQL::ExecutionError.new("Can't access submission")
-      @error = bad_argument_error
+  def run
+    submission = Submission.find_by(id: @arguments[:id])
+
+    unless submission
+      raise GraphQL::ExecutionError, 'Submission from ID Not Found'
     end
 
-    admin? || partner?
-  end
+    unless (
+             matching_user(submission, @arguments&.[](:session_id)) &&
```

Comment:
**suggestion:** please put the call to `matching_user(...)` in a named method (perhaps `submitter?`) so that this `unless` statement is easier to read.

Reply:
I don't think I fully understand what you mean🤔

Reply:
draft_in_process?(submission, @arguments&.[](:session_id))